fix: Direct users to the CUDA metapackage guides for compilation (#2416)
conda-forge/conda-forge.github.io: docs/maintainer/knowledge_base.md

## CUDA builds

Although the provisioned CI machines do not feature a GPU, conda-forge does provide mechanisms
to build CUDA-enabled packages.
See the [guide for maintainers of recipes that use CUDA](https://github.com/conda-forge/cuda-feedstock/blob/main/recipe/doc/recipe_guide.md)
for more information.

> If a feedstock does need access to additional resource (like GPUs), please see the following section (#packages-that-require-a-gpu-or-long-running-builds)

Author: Ah, `#packages-that-require-a-gpu-or-long-running-builds` didn't link properly. Let me fix that.

Author: Fixed now (preview at https://deploy-preview-2416--conda-forge-previews.netlify.app/docs/maintainer/knowledge_base/#cuda-builds), so this is ready for review again.

## CUDA builds

Although the provisioned CI machines do not feature a GPU, conda-forge does provide mechanisms
to build CUDA-enabled packages.
See the [relevant CUDA user guides](https://github.com/conda-forge/cuda-feedstock/blob/main/recipe/README.md)

> for more information.

Author: Rerequesting review from @conda-forge/core as there were significant changes to the knowledge base since the initial approval.

Reviewer: This seems reasonable

We could consider linking directly to the recipe guide

```suggestion
to build CUDA-enabled packages.
See the [relevant CUDA user guides]( https://github.com/conda-forge/cuda-feedstock/blob/main/recipe/doc/recipe_guide.md )
for more information.
```

> 

Reviewer: Maybe it is worth calling out that GPU machines are available for CI when needed

```suggestion

If a feedstock does need access to additional resource (like GPUs), please see the following section (#packages-that-require-a-gpu-or-long-running-builds)

```

## CUDA builds

Although the provisioned CI machines do not feature a GPU, conda-forge does provide mechanisms
to build CUDA-enabled packages.
See the [guide for maintainers of recipes that use CUDA](https://github.com/conda-forge/cuda-feedstock/blob/main/recipe/doc/recipe_guide.md)
for more information.

> If a feedstock does need access to additional resource (like GPUs), please see the following

Author: Thanks for catching this @kkraus14!

#### My feedstock is not building old CUDA versions anymore

As new CUDA versions come out regularly, periodically conda-forge needs to
decide how many versions will be supported within resource constraints.

> As of January 2025, conda-forge supports CUDA 11.8 and 12.

Reviewer: Maybe we could just do 11 and 12 then? We are supporting the latest minor version of each

```suggestion
As of January 2025, conda-forge supports CUDA 11 and 12.
```

2. In your feedstock fork, create a new branch and place the migration file under `.ci_support/migrations`.

> 3. Open a PR and re-render. CUDA 9.2, 10.0 and 10.1 will appear in the CI checks now. Merge when ready!

Reviewer: The above section at least in terms of a question makes sense. However the content no longer applies as we [dropped CUDA 11.2]( https://conda-forge.org/news/2024/03/06/dropping-cuda-112/ ) and earlier CUDA versions. At this point conda-forge is on [CUDA 11.8+]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/a5db2568f1bb32782ccebb9edf196343da9718d1/recipe/conda_build_config.yaml#L58-L62 )

Was thinking maybe we could rewrite it with something like this?

```markdown
As new CUDA versions come out regularly, periodically conda-forge needs to
decide how many versions will be supported within resource constraints.

As of this moment (January 2025), conda-forge supports CUDA 11.8 and 12.

Pulling in the latest versions can be done by [rerendering](
updating_pkgs.md#dev-update-rerender ). There may be other fixes needed for the
feedstock depending on when it was last updated.
```

Open to suggestions
